### PR TITLE
Upgrade to latest popper library (yes, we tried to test everything)

### PR DESCRIPTION
**Pull request by Discordius**
This moves us from Popper 1.x, to Popper 2.x, which enables a lot of small fixes. In particular it makes the flipping behavior more consistent. 

The biggest change to the user experience from making this update should be that when scrolling across elements that trigger tooltips on hover, hovers now reliably don't trigger (as opposed to triggering most of the time, but sometimes not triggering, depending on a bunch of really subtle factors on how you scroll). This is primarily downstream of a [Chrome update in late 2020](https://chromestatus.com/feature/5697181675683840) that deactivated hover events on scroll, though the old Popper was attaching a bunch of extra event handlers that accidentally partially restored the previous chrome behavior here. 

We tested every callsite of LWTooltip and LWPopper and tried to restore the previous behavior as much as possible, with some small changes in places where it seemed wrong.

We also simplified the API. Previously when you fed in modifiers into the old Popper implementation, it usually just deactivated the overflow and flipping behavior. We have now replaced all of those call sites with a straightforward "allowOverflow: true" argument, which seems to capture the intention better. If we want to access the more advanced modifiers of Popper 2.x, we can add additional arguments to LWPopper as needed.

### CODE REVIEW

**Review by Xodarap — APPROVED** [review]
I mostly tested forum specific stuff and hovers all still seem to work. :shipit:

File: packages/lesswrong/components/common/LWPopper.tsx [hunk 1]
```diff
@@ -29,38 +31,44 @@ const styles = (theme: ThemeType): JssStyles => ({
...
   
   return (
-    <Popper 
-      className={classNames(classes.popper, {[classes.noMouseEvents]: !clickable})} 
-      modifiers={newModifiers} 
-      open={open}
-      {...props}
-    >
-      <div className={classNames({[classes.tooltip]: tooltip, [classes.default]: !tooltip})}>
+    createPortal(
```
Comment (jpaddison3) [inline packages/lesswrong/components/common/LWPopper.tsx:61]: This needs a comment linking to their documentation

File: packages/lesswrong/components/common/LWPopper.tsx [hunk 2]
```diff
@@ -29,38 +31,44 @@ const styles = (theme: ThemeType): JssStyles => ({
 })
 
 // This is a thin wrapper over material-UI Popper so that we can set default z-index and modifiers
```
Comment (jpaddison3) [inline packages/lesswrong/components/common/LWPopper.tsx:33]: This comment is wrong

**Comment by jimrandomh** [comment]
Tried this out in a pair-programming session with JP, particularly with respect to #4950 . This upgrade does not fix the problem. Navigation sidebar links are still unclickable sometimes, based on subtleties of tooltip dimensions and window size. Reproduced the EA Forum Library link being unclickable, on Browserstack with an iPhone 13.

I believe both the old and new versions of the `popperjs` library are fundamentally built on the wrong abstractions, in such a way that this class of bug can't be fixed without removing our reliance on it.

**Pull request merged**